### PR TITLE
outguess: init at 0.4

### PR DESCRIPTION
**Pull request by HeitorAugustoLN**
## Description of changes

- Add outguess package

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by numinit** [comment]
@ofborg build outguess

**Comment by numinit** [comment]
Nice! Thought about packaging this myself, glad someone did it.

**Review by numinit — APPROVED** [review]

File: pkgs/by-name/ou/outguess/package.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  nativeBuildInputs = [ autoconf automake ];
+
+  buildPhase = ''
+    ./autogen.sh
+    ./configure --with-generic-jconfig --prefix=$out
+    make
+  '';
+
+  installPhase = ''
+    make install
```
Comment (Aleksanaa) [inline pkgs/by-name/ou/outguess/package.nix:27]: This should be ran automatically.

File: pkgs/by-name/ou/outguess/package.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  '';
+
+  meta = with lib; {
+    description = "Outguess is a universal steganographic tool that allows the insertion of hidden information into the redundant bits of data sources.";
```
Comment (Aleksanaa) [inline pkgs/by-name/ou/outguess/package.nix:31]:
```suggestion
    description = "Universal steganographic tool that allows the insertion of hidden information into the redundant bits of data sources";
```

File: pkgs/by-name/ou/outguess/package.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  pname = "outguess";
+  version = "0.4";
+  src = fetchFromGitHub {
+    owner = "resurrecting-open-source-projects";
+    repo = "outguess";
+    rev = version;
+    hash = "sha256-yv01jquPTnVk9fd1tqAt1Lxis+ZHZqdG3NiTFxfoXAE=";
+  };
+
+  nativeBuildInputs = [ autoconf automake ];
```
Comment (Aleksanaa) [inline pkgs/by-name/ou/outguess/package.nix:18]:
```suggestion
  nativeBuildInputs = [ autoReconfHook ];
```

Try remove buildPhase, should work

File: pkgs/by-name/ou/outguess/package.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
+{ fetchFromGitHub
+, stdenv
+, lib
+, autoconf
+, automake
+}:
+
+stdenv.mkDerivation rec {
+  pname = "outguess";
+  version = "0.4";
```
Comment (Aleksanaa) [inline pkgs/by-name/ou/outguess/package.nix:10]:
```suggestion
  version = "0.4";

```

File: pkgs/by-name/ou/outguess/package.nix [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+  installPhase = ''
+    make install
+  '';
+
+  meta = with lib; {
+    description = "Outguess is a universal steganographic tool that allows the insertion of hidden information into the redundant bits of data sources.";
+    homepage = "https://github.com/resurrecting-open-source-projects/outguess";
+    license = licenses.bsdOriginal;
+    maintainers = with maintainers; [ HeitorAugustoLN ];
```
Comment (Aleksanaa) [inline pkgs/by-name/ou/outguess/package.nix:34]:
```suggestion
    maintainers = with maintainers; [ HeitorAugustoLN ];
    mainProgram = "<main executable?>";
    platforms = platforms.unix;
```

File: pkgs/by-name/ou/outguess/package.nix [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    ./autogen.sh
+    ./configure --with-generic-jconfig --prefix=$out
+    make
+  '';
+
+  installPhase = ''
+    make install
+  '';
+
+  meta = with lib; {
```
Comment (Aleksanaa) [inline pkgs/by-name/ou/outguess/package.nix:30]:
```suggestion
  meta = {
```

Better to remove it.

Comment (HeitorAugustoLN) [reply pkgs/by-name/ou/outguess/package.nix]: Removed.

File: pkgs/by-name/ou/outguess/package.nix (hunk 3 above)
Comment (HeitorAugustoLN) [reply pkgs/by-name/ou/outguess/package.nix]: It worked, just needed to add `configureFlags = [ "--with-generic-jconfig" ];`

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**